### PR TITLE
Dotty progress

### PR DESCRIPTION
**Pull request by m-sp**
misc dotty fixes

### CODE REVIEW

File: circe/src/test/scala/org/http4s/circe/CirceSuite.scala [hunk 1]
```diff
@@ -77,9 +77,7 @@ class CirceSuite extends JawnDecodeSupportSuite[Json] with Http4sLawSuite {
 
   test("json encoder should have json content type") {
     val maybeHeaderT: Option[`Content-Type`] = jsonEncoder[IO].headers.get(`Content-Type`)
-    assertEquals(
-      maybeHeaderT,
-      Some(`Content-Type`(MediaType.application.json)))
+    assertEquals(maybeHeaderT, Some(`Content-Type`(MediaType.application.json)))
```
Comment (rossabaker) [inline circe/src/test/scala/org/http4s/circe/CirceSuite.scala:80]: Did scalafmt want these?  I don't see the difference.

**Comment by rossabaker** [comment]
Oh, I was somehow just looking at the last commit!

**Comment by rossabaker** [comment]
Okay, I understand now. :+1:

File: circe/src/test/scala/org/http4s/circe/CirceSuite.scala (hunk 1 above)
Comment (m-sp) [reply circe/src/test/scala/org/http4s/circe/CirceSuite.scala:80]: Nah it works around inference issues. Would be solved with the header v2s aswell just wanted to make some progress

**Pull request merged**